### ISSUE TITLE
elasticsearch not listening port 9200

### ISSUE THREAD

**Issue by dashxdr** [AlphaReign/scraper#17]
% sudo apt-get install elasticsearch
Reading package lists... Done
Building dependency tree       
Reading state information... Done
elasticsearch is already the newest version (1.7.3+dfsg-3).
0 upgraded, 0 newly installed, 0 to remove and 48 not upgraded.
% sudo /etc/init.d/elasticsearch start
[ ok ] Starting elasticsearch (via systemctl): elasticsearch.service.
% netstat -a -n | grep 9200
%

**Comment by Raxvis** [comment]
what does ```curl 127.0.0.1:9200``` give you?

what does ```top``` show?

what does ```service elasticsearch status``` show?

**Comment by dashxdr** [comment]
% curl 127.0.0.1:9200
curl: (7) Failed to connect to 127.0.0.1 port 9200: Connection refused

TOP: the relevant line you're interested in:
 7891 dash      20   0 1237512  92896  16584 S  13.3  0.3   4:23.86 node /ram/scrap                                                                             

% service elasticsearch status
● elasticsearch.service - LSB: Starts elasticsearch
   Loaded: loaded (/etc/init.d/elasticsearch; bad; vendor preset: enabled)
   Active: active (exited) since Tue 2018-04-03 14:52:25 CDT; 2min 17s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 9764 ExecStop=/etc/init.d/elasticsearch stop (code=exited, status=0/SUCCESS)
  Process: 9798 ExecStart=/etc/init.d/elasticsearch start (code=exited, status=0/SUCCESS)

Apr 03 14:52:25 DaveDesktop systemd[1]: Starting LSB: Starts elasticsearch...
Apr 03 14:52:25 DaveDesktop systemd[1]: Started LSB: Starts elasticsearch.

**Comment by dashxdr** [comment]
I'm struggling to construct a viable /etc/elasticsearch/elasticsearch.yml file, but this isn't good enough:

cluster.name: mycluster1
node.name: "My First Node"
http.port: 9200
network.bind_host: 127.0.0.1
node.master: true
node.data: true
http.enabled: true

**Comment by ghost** [comment]
Are you running this locally or on a server? Check to make sure 9200 port is open ?

**Comment by Raxvis** [comment]
what does ```sudo journalctl --unit elasticsearch``` show?

**Comment by dashxdr** [comment]
Locally on my desktop. Nothing's listening on 9200, no conflicts.
Apr 03 15:01:05 DaveDesktop systemd[1]: Stopping LSB: Starts elasticsearch...
Apr 03 15:01:05 DaveDesktop systemd[1]: Stopped LSB: Starts elasticsearch.
Apr 03 15:01:05 DaveDesktop systemd[1]: Starting LSB: Starts elasticsearch...
Apr 03 15:01:05 DaveDesktop systemd[1]: Started LSB: Starts elasticsearch.
Apr 03 15:01:54 DaveDesktop systemd[1]: Stopping LSB: Starts elasticsearch...
Apr 03 15:01:54 DaveDesktop systemd[1]: Stopped LSB: Starts elasticsearch.
Apr 03 15:01:54 DaveDesktop systemd[1]: Starting LSB: Starts elasticsearch...
Apr 03 15:01:54 DaveDesktop systemd[1]: Started LSB: Starts elasticsearch.

**Comment by Raxvis** [comment]
Do you have java installed?

Are there any logs here: ```/var/logs/elasticsearch```

**Comment by dashxdr** [comment]
% ls /var/log/elasticsearch/
% 

Nothing

**Comment by Raxvis** [comment]
what does ```which java``` show?

**Comment by dashxdr** [comment]
https://discuss.elastic.co/t/cant-start-elasticsearch-with-ubuntu-16-04/48730/8
It seems that to get Elasticsearch to run on 16.04 you have to set START_DAEMON to true on /etc/default/elasticsearch. It comes commented out by default, and uncommenting it makes Elasticsearch start again just fine.

**Comment by Raxvis** [comment]
Good to hear

**Comment by dashxdr** [comment]
% which java
/usr/bin/java
% 

elasticsearch running, here's what 127.0.0.1:9200 shows:

status | 200
name | "My First Node"
cluster_name | "mycluster1"
version |  
number | "1.7.3"
build_hash | "NA"
build_timestamp | "NA"
build_snapshot | false
lucene_version | "4.10.4"
tagline | "You Know, for Search"

**Comment by Raxvis** [comment]
That's exactly what you should see.  Check your pm2 logs and you should see it connected (or not erroring anymore)

FYI, running ```pm2 flush``` will clear out the old bad logs

**Comment by dashxdr** [comment]
0|index    | UDP Server listening on 0.0.0.0:6881

How do I do a search then?

**Comment by Raxvis** [comment]
Originally this was on a website (code here: https://github.com/AlphaReign/www-php)

If this is for local use, you can curl the data straight from ElasticSearch

Here is the code that constructed the search object for ElasticSearch on the site
https://github.com/AlphaReign/www-php/blob/master/handlers/search/query.php

**Comment by ghost** [comment]
Not sure why you helping him when he got so pissed in the last issue haha

**Comment by Raxvis** [comment]
Mostly so that when others come and have questions, they will get answers.  It's more like building a knowledge base.

And honestly, if I saw enough activity on this.  I might pick it back up and make it a nice self contained system and actually create something more from it.  I was already testing building it into a single app for people to run, but I want to actually create a way for people to setup nodes and then users can just install an app that connects to those nodes and searches across them (and maybe event downloads the torrent files).  It's been in the back of my mind for a long time, but essentially, it's a completely resistant network that can't be taken down like other torrent sites.

**Comment by dashxdr** [comment]
Maybe if a person gets pissed it's not automatically 100% their fault?

Re the query.php reference can you provide an example query as in
http://127.0.0.1:9200/?search=Red
I don't mind bypassing the www-php front end. My goal is to run a DHT node and just be able to query the accumulated magnet links. There was no need to do this until the nice alphareign website went away (Thanks for keeping it up for so long, btw!).
ETA: My bad, I mixed up skytorrents and alphareign.

**Comment by Raxvis** [comment]
When you get pissed right out the gate, it's a little hard to make it someone else's fault.

Here is an example query.  I don't know if it works on the latest version of ElasticSearch

```
POST http://127.0.0.1/_search

{
    "query": {
        "function_score": {
            "boost_mode": "sum",
            "field_value_factor": {
                "factor": 0.1,
                "field": "seeders",
                "missing": 0,
                "modifier": "sqrt"
            },
            "max_boost": 10,
            "query": {
                "bool": {
                    "must": {
                        "match": {
                            "search": {
                                "operator": "and",
                                "query": "red"
                            }
                        }
                    },
                    "must_not": [
                        {
                            "term": {
                                "inactive": true
                            }
                        },
                        {
                            "range": {
                                "flags": {
                                    "gte": 5
                                }
                            }
                        }
                    ]
                }
            }
        }
    }
}
```

**Comment by dashxdr** [comment]
http://127.0.0.1:9200/_search
That got something.
http://okfnlabs.org/blog/2013/07/01/elasticsearch-query-tutorial.html#curl-or-browser

**Comment by dashxdr** [comment]
"When you get pissed right out the gate, it's a little hard to make it someone else's fault."

It was only out of the gate from the point of view of github. What led up to the frustration? I don't claim to not being an asshole. Sometimes being an asshole works. I do claim I'm not a troll.

"Mostly so that when others come and have questions, they will get answers. It's more like building a knowledge base."

Exactly right.

**Comment by dashxdr** [comment]
I don't know if it's relevant but as the scraper is running I keep getting error messages:
```
0|index    | Queue Processed
0|index    | { [Error: ActionRequestValidationException[Validation Failed: 1: script or doc is missing;]]
0|index    |   status: 400,
0|index    |   displayName: 'BadRequest',
0|index    |   message: 'ActionRequestValidationException[Validation Failed: 1: script or doc is missing;]' }
0|index    | Queuing: 0aeda8d2fbe299a1a1f8914ea330b137b41c2c6e | Power ISO 5.8 + Serial Keys Working maC

```

**Comment by Raxvis** [comment]
A 400 error indicates that the body your sending to ES is malformed.

Does it show that after every ```Queue Processed``` line?

**Comment by dashxdr** [comment]
That's the output of "pm2 logs 0" and it seems to be related to the DHT activity... I'm not doing anything to trigger it.

**Comment by Raxvis** [comment]
Right.  I mean, if you watch the logs, do you always see the ```status: 400`` after ```Queue Processed```

**Comment by dashxdr** [comment]
Yes, I only see the Error 400 after "Queue Processed".  I never see the "Creation Times Added". I'm assuming this is coming from src/indexer.js which I'm trying to fathom...

localhost:9200/_search has 390 hits reported. It's steadily rising. UDP on port 6881 is going like gangbusters though. I only know in general terms what DHT is doing, not the nitty gritty.

**Comment by Raxvis** [comment]
I am guessing you are getting errors with having bad update documents trying to be inserted.

As long as records are being inserted, you should be fine.  The bulk load will load all it can and fail the ones that aren't correct

**Comment by dashxdr** [comment]
dist/indexer.js I added a console.log(creations) after the console.log('Queue Processed');
```
0|index    | Queuing: f4d4c5d164d6e7614937116f5698a4ba1895eaa7 | Mollys Game 2017
0|index    | Queue Processed
0|index    | [ { update: 
0|index    |      { _id: 'f4d4c5d164d6e7614937116f5698a4ba1895eaa7',
0|index    |        _index: 'torrents',
0|index    |        _retry_on_conflict: 3,
0|index    |        _type: 'hash' } },
0|index    |   { lang: 'painless',
0|index    |     script: 
0|index    |      { inline: 'if( !ctx._source.containsKey("created") ){ ctx._source.created = params.time; }',
0|index    |        params: [Object] } } ]
0|index    | { [Error: ActionRequestValidationException[Validation Failed: 1: script or doc is missing;]]
0|index    |   status: 400,
0|index    |   displayName: 'BadRequest',
0|index    |   message: 'ActionRequestValidationException[Validation Failed: 1: script or doc is missing;]' }

```
It may be elasticsearch requires the script to be inside the update object or something... might be their API changed since you wrote the original code. Regardless... my goal was to get a taste of DHT scraping / harvesting and I think that is accomplished (Thanks!).

**Comment by Raxvis** [comment]
looks like the script is wrong which is fine.  Torrents are still being loaded in.  Just not adding the created time

**Comment by dashxdr** [comment]
Might have fixed it. Modified dist/indexer.js the script declaration:
```
			var script = {
				lang: 'painless',
				script: 'if( !ctx._source.containsKey("created") ){ ctx._source.created = params.time; }',
				params: { time: Math.floor(Date.now() / 1000) },
			};

```
The error doesn't appear anymore
0|index    | Queue Processed
0|index    | Creation Times Added

**Comment by Raxvis** [comment]
Awesome.  It looks like they changed the script layout versus when I had it last running

**Comment by dashxdr** [comment]
Probably a similiar fix for src/indexer.js would work but I didn't bother. See pull request.

**Comment by Raxvis** [comment]
Thanks for the pull request.  I pulled it in.  I also brought in the changes to src

**Comment by ghost** [comment]
Hey prefinem did you ever get chance to add in my fix for the file size bug ?

Kind regards

**Comment by Raxvis** [comment]
I had it, but for some reason I can't find it.  Could you resend it and I will pull it in.

**Comment by ghost** [comment]
https://github.com/AlphaReign/scraper/files/1804834/src_dist_index_js.zip

There you go :)

**Comment by Raxvis** [comment]
Merged as part of #19

Thanks!
